game-core/src/main/java/games/strategy/util/PointFileReaderWriter.java

```diff
@@ -188,6 +188,69 @@ public static void writeOneToMany(final OutputStream sink, final Map<String, Lis
     return mapping;
   }
 
+  /**
+   * Writes the specified one-to-many mapping between names and (points, overflowToLeft) to the specified stream.
+   *
+   * @param sink The stream to which the name-to-points mappings will be written.
+   * @param mapping The name-to-points mapping to be written.
+   *
+   * @throws IOException If an I/O error occurs while writing to the stream.
+   */
+  public static void writeOneToManyPlacements(final OutputStream sink,
+      final Map<String, Tuple<List<Point>, Boolean>> mapping)
+      throws IOException {
+    checkNotNull(sink);
+    checkNotNull(mapping);
+
+    final StringBuilder out = new StringBuilder();
+    final Iterator<String> keyIter = mapping.keySet().iterator();
+    while (keyIter.hasNext()) {
+      final String name = keyIter.next();
+      out.append(name).append(" ");
+      final Collection<Point> points = mapping.get(name).getFirst();
+      final boolean overflowToLeft = mapping.get(name).getSecond();
+      final Iterator<Point> pointIter = points.iterator();
+      while (pointIter.hasNext()) {
+        final Point point = pointIter.next();
+        out.append(" (").append(point.x).append(",").append(point.y).append(")");
+        if (pointIter.hasNext()) {
+          out.append(" ");
+        }
+      }
```

Comment:
This is a really good use case for Streams:
```java
out.append(points.stream()
  .map(point -> " (" + point.x + "," + point.y + ") ")
  .collect(Collectors.joining(" ")));
```

Reply:
Left it this way for consistency with other similar methods in this class. Probably updating this class to use streams in a number of places as a separate PR would be better.

Reply:
Wait, so does that mean we have a DRY violation? Why do we have such similar methods at all? When adding a new variation, should we not have updated the internal design and extracted the common bits?

I don't know if everyone will always see all the different instances and realize that they are *supposed* to be consistent.